include/openssl/ssl.h

```diff
@@ -586,12 +590,12 @@ typedef int (*SSL_async_callback_fn)(SSL *s, void *arg);
  * cannot be used to clear bits.
  */
 
-unsigned long SSL_CTX_get_options(const SSL_CTX *ctx);
-unsigned long SSL_get_options(const SSL *s);
-unsigned long SSL_CTX_clear_options(SSL_CTX *ctx, unsigned long op);
-unsigned long SSL_clear_options(SSL *s, unsigned long op);
-unsigned long SSL_CTX_set_options(SSL_CTX *ctx, unsigned long op);
-unsigned long SSL_set_options(SSL *s, unsigned long op);
+uint64_t SSL_CTX_get_options(const SSL_CTX *ctx);
+uint64_t SSL_get_options(const SSL *s);
+uint64_t SSL_CTX_clear_options(SSL_CTX *ctx, uint64_t op);
+uint64_t SSL_clear_options(SSL *s, uint64_t op);
+uint64_t SSL_CTX_set_options(SSL_CTX *ctx, uint64_t op);
+uint64_t SSL_set_options(SSL *s, uint64_t op);
```

Comment:
I wonder if these signature changes will result in any compilation issues for end user applications.

Reply:
For (most) 64-bit platforms, it shouldn't make much of a difference; assuming **unsigned long** ~ **uint64_t**
For 32-bit platforms, there will be some issues, and users would have to cast them away, or handle the return value properly. Since all the values are in the lower 32-bits, casting won't have issues until more options are used.

Reply:
It's unclear that this counts as "API compatible; e.g., I expect modern compilers to warn about the truncated return value.

Reply:
Yes, compiles will warn, but it won't break anything until upper-32-bit option values are defined.

Reply:
I think this warrants a discussion on openssl-project

Reply:
I started to write an email to the list about this, but in putting it together I think I answered the question I was going to ask. Specificially openssl/web#82 says:

"No existing public interface can be modified except where changes are unlikely to break source compatibility."

On thinking about this I think a warning is likely to be sufficient to break some builds. Many builds treat warnings as errors. So I'm thinking that this change is probably a step too far.

My suggestion is to remove it from this PR - since really this is just an incidental tangent to the main point of this PR. If we are only targetting 3.0 then we can reuse one of the currently "reserved" SSL_OP_ values for now. We should probably also raise an issue that we're running out of SSL_OP_ values and we can figure out in a separate PR the best way of handling it.

Reply:
Thanks (I should've come back to look at this earlier). Certainly have a 64-suffix version would allow for extension. This would only be an issue with 32-bit platforms. Let's say we drop those. ;) /s

Reply:
Indeed, if we are only targeting 3.0.0, then we are fine re-using the bit. But, I do see a use for adding it to 1.1.1, but not critically.

Reply:
> This would only be an issue with 32-bit platforms. Let's say we drop those.

Technically those with a 32-bit long. This includes 64-bit Windows. :-(

Reply:
I was being sarcastic! :) But I undid the 64-bit changes.